openedx_tagging/core/tagging/models/base.py

```diff
@@ -404,6 +404,20 @@ def _find_object_tag_index(tag_ref, object_tags) -> int:
                 -1,
             )
 
+        def _check_current_tag_count() -> None:
+            """
+            Checks if the current count of tags for the object is less than 100
+            """
+            # Exclude self.id to avoid counting the tags that are going to be updated
+            current_count = ObjectTag.objects.filter(object_id=object_id).exclude(taxonomy_id=self.id).count()
+
+            if current_count >= 100:
+                raise ValueError(
+                    _(f"Object ({object_id}) already have 100 or more tags.")
+                )
+
+        _check_current_tag_count()
```

Comment:
Shouldn't this check also include the count of the new tags that will will be potentially be added, since this check is happening at the beginning of the function call? Eg: What happens when the `current_count` is say `99` but there are `2` _new_ tags in the `tags` list leading to `101`, would this check catch it?

Reply:
You're right @yusuf-musleh! I totally missed the allow_multiple cases.
Thank you for pointing out!

Reply:
Done 432e4c22eb18a05eeafb5070a14666680f4c1484